src/main/java/dev/jbang/source/sources/JavaSource.java

```diff
@@ -25,7 +27,9 @@ protected JavaSource(ResourceRef ref, String script, Function<String, String> re
 
 	@Override
 	protected List<String> getCompileOptions() {
-		return tagReader.collectOptions("JAVAC_OPTIONS", "COMPILE_OPTIONS");
+		List<String> gopts = Collections.singletonList("-g");
```

Comment:
not really a fan of adding this here - won't this mean that if you have 10 source files you get 10 x "-g" ?

Reply:
Possibly, yes. For now I just don't know where else to put it :-/
For some time now I've thought that there should probably be a class for each kind of project, eg JavaProject, KotlinProject, etc. Then we'd have a proper place to put things like that.

Reply:
We can of course leave it where it is and surround it with an instanceof test to see if the main source file is JavaSource or not.

Reply:
maybe have a `LinkedSet contributeGlobalCompileOptions();` which allows a Source to add to a set that gets added first?